src/storage/storage.cpp

```diff
@@ -184,643 +182,670 @@ Storage::ReturnCode Storage::Run(int max_wait)
-    {
-        std::copy(datasource_names_data.names.begin(),
-                  datasource_names_data.names.end(),
-                  datasource_name_data_ptr);
+        const auto turn_lane_mask_ptr =
+            layout_ptr->GetBlockPtr<extractor::guidance::TurnLaneType::Mask, true>(
+                memory_ptr, DataLayout::LANE_DESCRIPTION_MASKS);
+        if (!lane_description_masks.empty())
+        {
+            BOOST_ASSERT(layout_ptr->GetBlockSize(DataLayout::LANE_DESCRIPTION_MASKS) >=
+                         sizeof(lane_description_masks[0]) * lane_description_masks.size());
+            std::copy(
+                lane_description_masks.begin(), lane_description_masks.end(), turn_lane_mask_ptr);
+        }
     }
 
-    auto datasource_name_offsets_ptr = shared_layout_ptr->GetBlockPtr<std::uint32_t, true>(
-        shared_memory_ptr, SharedDataLayout::DATASOURCE_NAME_OFFSETS);
-    if (shared_layout_ptr->GetBlockSize(SharedDataLayout::DATASOURCE_NAME_OFFSETS) > 0)
+    // Load original edge data
     {
-        std::copy(datasource_names_data.offsets.begin(),
-                  datasource_names_data.offsets.end(),
-                  datasource_name_offsets_ptr);
+        io::File edges_input_file(config.edges_data_path);
+
+        const auto number_of_original_edges = edges_input_file.readElementCount64();
+
+        const auto via_geometry_ptr =
+            layout_ptr->GetBlockPtr<GeometryID, true>(memory_ptr, DataLayout::VIA_NODE_LIST);
+
+        const auto name_id_ptr =
+            layout_ptr->GetBlockPtr<unsigned, true>(memory_ptr, DataLayout::NAME_ID_LIST);
+
+        const auto travel_mode_ptr = layout_ptr->GetBlockPtr<extractor::TravelMode, true>(
+            memory_ptr, DataLayout::TRAVEL_MODE);
+        const auto pre_turn_bearing_ptr =
+            layout_ptr->GetBlockPtr<util::guidance::TurnBearing, true>(
+                memory_ptr, DataLayout::PRE_TURN_BEARING);
+        const auto post_turn_bearing_ptr =
+            layout_ptr->GetBlockPtr<util::guidance::TurnBearing, true>(
+                memory_ptr, DataLayout::POST_TURN_BEARING);
+
+        const auto lane_data_id_ptr =
+            layout_ptr->GetBlockPtr<LaneDataID, true>(memory_ptr, DataLayout::LANE_DATA_ID);
+
+        const auto turn_instructions_ptr =
+            layout_ptr->GetBlockPtr<extractor::guidance::TurnInstruction, true>(
+                memory_ptr, DataLayout::TURN_INSTRUCTION);
+
+        const auto entry_class_id_ptr =
+            layout_ptr->GetBlockPtr<EntryClassID, true>(memory_ptr, DataLayout::ENTRY_CLASSID);
+
+        io::readEdges(edges_input_file,
+                      via_geometry_ptr,
+                      name_id_ptr,
+                      turn_instructions_ptr,
+                      lane_data_id_ptr,
+                      travel_mode_ptr,
+                      entry_class_id_ptr,
+                      pre_turn_bearing_ptr,
+                      post_turn_bearing_ptr,
+                      number_of_original_edges);
```

Comment:
can we add some `i/o iterator` to the file and do a ranged based for loop for stuff like this?